interfaces/bspline_vector_space.py

```diff
@@ -13,11 +13,10 @@ class BsplineVectorSpace(VectorSpace):
     def __init__(self, degree=0, knots=[]):
         """ Pure interface class. It generates the constant on [0,1] if no arguments are provided. """
         assert degree >= 0
-        assert len(knots) > 1
         assert knots[0] != knots[-1]
```

Comment:
move this below, otherwise you'll get an assertion when trying to access an empty knot list